lib/npg_qc/autoqc/role/generic.pm

```diff
@@ -17,10 +16,40 @@ around 'filename_root' => sub {
   return join q[.], $self->$orig(), $self->pp_name || q[unknown];
 };
 
-sub doc2yaml {
+sub massage_for_render {
   my $self = shift;
-  my $doc = $self->doc;
-  return $doc ? Dump($doc) : q[];
+
+  my %table_data = ();
+  if (exists $self->doc->{'QC summary'}) {
+    my $qc_data = $self->doc->{'QC summary'};
+
+    foreach my $allowed_key (qw/
+      longest_no_N_run pct_N_bases qc_pass max_negative_control_filtered_read_count
+    /) {
+      if (exists $qc_data->{$allowed_key}) {
+        $table_data{$allowed_key} = $qc_data->{$allowed_key};
+      } else {
+        $table_data{$allowed_key} = q();
+      }
+    }
+    # rename fields to improve viewer comprehension
+    $table_data{num_aligned_fragments} = $qc_data->{num_aligned_reads};
+  }
+  if (exists $self->doc->{meta}) {
+    # All current analyses that use this particular data are paired-end
+    # and as such, the input fragment count does not reflect it.
+    # Single-ended fragment counts will be incorrect!
+    $table_data{num_input_fragments} = $self->doc->{meta}{num_input_reads} * 2;
```

Comment:
Need to check that $self->doc->{meta}{num_input_reads} is defined before multiplying it. Something like
if (defined $self->doc->{meta}{num_input_reads}) {
  $table_data{num_input_fragments} = $self->doc->{meta}{num_input_reads} * 2;
}

Reply:
Is everything in meta optional?